### PR TITLE
✨ feat (App Frontend): Loading Spinners for Buttons

### LINKED ISSUE #21: Add loading spinners to buttons on preference selection modal

### LINKED ISSUE #29: Indicate too large delegations in Role Application Selection Process
If the Delegation is too large for any option to choose from, instead of being just an empty table it should tell the head delegate that the delegation is too large and cannot participate in the selection process this way.

### PR DESCRIPTION
**Pull request by TadeSF**
branch: 21-add-loading-spinners-to-buttons-on-preference-selection-modal

closes #21
closes #29

### CODE REVIEW

File: src/lib/components/SquareButtonWithLoadingState.svelte [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+		await onClick();
+		loading = false;
+	};
+</script>
+
+<button class="btn btn-square btn-sm {cssClass}" onclick={fullOnClick} {disabled}>
+	{#if loading}
+		<i class="fa-{duotone ? 'duotone' : 'solid'} fa-spinner fa-spin"></i>
+	{:else}
+		<i class="fa-{duotone ? 'duotone' : 'solid'} fa-{icon.replace('fa-', '')}"></i>
```
Comment (m1212e) [inline src/lib/components/SquareButtonWithLoadingState.svelte:25]: Just curious: Why the replace call? :D

File: src/routes/(authenticated)/dashboard/[conferenceId]/DelegationRegistrationStage.svelte [hunk 2]
```diff
@@ -122,7 +123,7 @@
 				newHeadDelegateUserId: userId
 			})
 		);
-		invalidateAll();
+		await invalidateAll();
```
Comment (m1212e) [inline src/routes/(authenticated)/dashboard/[conferenceId]/DelegationRegistrationStage.svelte:126]: We should consider using fine grained invalidate calls for faster updates: https://kit.svelte.dev/docs/modules#$app-navigation-invalidate but this is fine for now

**Review by m1212e — APPROVED** [review]

File: src/lib/components/SquareButtonWithLoadingState.svelte (hunk 1 above)
Comment (TadeSF) [reply src/lib/components/SquareButtonWithLoadingState.svelte:25]: I always confuse where I implemented the fontAwesome icon to be "fa-icon" or just "icon" – in other words, does the css class need the "fa-" prefix or is it already present. When you copy from the fontawesome website, it does not provide you with the prefix, but being so used to it I often include it anyways which breaks the icon.

In order to fix this while not trusting myself to actually remember how to do it, I thought having the prefix inside the class and omiting the "fa-" if present in the icon-string, I would solve this problem for good. So now: wherever I implement Icons in components, where the icons are not fixed but rather a prop, I implement it this way :D

File: src/routes/(authenticated)/dashboard/[conferenceId]/DelegationRegistrationStage.svelte (hunk 2 above)
Comment (TadeSF) [reply src/routes/(authenticated)/dashboard/[conferenceId]/DelegationRegistrationStage.svelte:126]: Yes, did stumble across that as well. Goes into the chapter performance optimization and telemetry

**Pull request merged**